DifferenceGenerator/src/main/kotlin/DifferenceGenerator.kt

```diff
@@ -173,12 +183,43 @@ class DifferenceGenerator(
      * @param color the color
      * @return a Buffered Imnage colored in the given color
      */
-    private fun getColoredBufferedImage(color: Color): BufferedImage {
-        val result = BufferedImage(width, height, BufferedImage.TYPE_3BYTE_BGR)
+    private fun getColoredBufferedImage(
+        color: Color,
+        type: Int = BufferedImage.TYPE_3BYTE_BGR,
+    ): BufferedImage {
+        val result = BufferedImage(width, height, type)
         val g2d: Graphics2D = result.createGraphics()
         g2d.paint = color
         g2d.fillRect(0, 0, width, height)
         g2d.dispose()
         return result
     }
+
+    private fun generateMasking() {
+        if (maskFile == null) {
+            mask = getColoredBufferedImage(Color(255, 255, 255, 0), BufferedImage.TYPE_4BYTE_ABGR)
+            return
+        }
+        val maskGrabber = ImageIO.read(maskFile)
+        if (maskGrabber.width != width || maskGrabber.height != height) {
+            throw Exception("Mask must have the same dimensions as the videos")
+        }
+        mask = BufferedImage(width, height, BufferedImage.TYPE_4BYTE_ABGR)
```

Comment:
Whats is the reason for using TYPE_4BYTE_ABGR here instead of the 3BYTE one we are already using?

Reply:
since i use graphics to draw the mask on top of frames i need some pixels to be transparent (the white ones) this should be alot faster than manual looping

Reply:
i use it twice and as mentioned below the weird replacement of pixels should be removed if we use a seethrough pixel in the masking image instead of white. That means it should be used only once as a backup if no mask is given